--- config-top.original.h	2017-06-08 11:01:00.595785218 +0000
+++ config-top.h	2017-06-08 11:01:23.255553826 +0000
@@ -113,7 +113,7 @@
 
 /* Define if you want each line saved to the history list in bashhist.c:
    bash_add_history() to be sent to syslog(). */
-/* #define SYSLOG_HISTORY */
+#define SYSLOG_HISTORY 
 #if defined (SYSLOG_HISTORY)
 #  define SYSLOG_FACILITY LOG_USER
 #  define SYSLOG_LEVEL LOG_INFO
